extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js

```diff
@@ -216,235 +140,257 @@ function tests(name) {
-        const secondLayout = ad.implementation_.layoutCallback();
-        expect(firstLayout).to.equal(secondLayout);
+      const firstLayout = ad.implementation_.layoutCallback();
+      const secondLayout = ad.implementation_.layoutCallback();
+      expect(firstLayout).to.equal(secondLayout);
 
-        ad.implementation_.unlayoutCallback();
-        const newLayout = ad.implementation_.layoutCallback();
-        expect(newLayout).to.not.equal(secondLayout);
+      ad.implementation_.unlayoutCallback();
+      const newLayout = ad.implementation_.layoutCallback();
+      expect(newLayout).to.not.equal(secondLayout);
+    });
+  });
+
+  describe('get CID', () => {
+    it('should propagete CID to ad iframe', () => {
+      sandbox.stub(adCid, 'getAdCid', () => {
+        return Promise.resolve('sentinel123');
+      });
+      return getAd().then(ad => {
+        const src = ad.firstChild.getAttribute('src');
+        expect(src).to.contain('"clientId":"sentinel123"');
       });
     });
 
-    describe('get CID', () => {
-      it('should propagete CID to ad iframe', () => {
-        sandbox.stub(adCid, 'getAdCid', () => {
-          return Promise.resolve('sentinel123');
-        });
-        return getAd().then(ad => {
-          const src = ad.firstChild.getAttribute('src');
-          expect(src).to.contain('"clientId":"sentinel123"');
-        });
+    it('should proceed w/o CID', () => {
+      sandbox.stub(adCid, 'getAdCid', () => {
+        return Promise.resolve(undefined);
       });
+      return getAd().then(ad => {
+        const src = ad.firstChild.getAttribute('src');
+        expect(src).to.contain('"clientId":null');
+      });
+    });
+  });
 
-      it('should proceed w/o CID', () => {
-        sandbox.stub(adCid, 'getAdCid', () => {
-          return Promise.resolve(undefined);
+  describe('no content handler', () => {
+    it('should display fallback when there is one and hide placeholder', () => {
+      return getAd(undefined, undefined, ad => {
+        const fallback = createElementWithAttributes(document, 'div', {
+          fallback: '',
         });
-        return getAd().then(ad => {
-          const src = ad.firstChild.getAttribute('src');
-          expect(src).to.contain('"clientId":null');
+        const placeholder = createElementWithAttributes(document, 'div', {
+          placeholder: '',
         });
+        ad.appendChild(fallback);
+        ad.appendChild(placeholder);
+        expect(placeholder.classList.contains('amp-hidden')).to.be.false;
+        return ad;
+      }).then(ad => {
+        sandbox.stub(
+            ad.implementation_, 'deferMutate', function(callback) {
+              callback();
+            });
+        expect(ad).to.not.have.class('amp-notsupported');
+        const placeholderEl = ad.querySelector('[placeholder]');
+
+        ad.implementation_.noContentHandler_();
+        expect(ad).to.have.class('amp-notsupported');
```

Comment:
This only works after `toggleFallback()` is called. https://github.com/ampproject/amphtml/blob/master/src/custom-element.js#L1262
Can't apply to line 192 because we haven't `#toggleFallback` then.